### PR TITLE
Refactors species stomach code, makes ethereal charge inherent to their stomach

### PR DESCRIPTION
**Pull request by Time-Green**
A stomach would literally just ask the species datum how to be hungry and made a bunch of exceptions for ethereals. I mostly just moved code around so that hunger is handled by a persons stomach, thus shrinking the species.dm bloat. 

This also makes it so ethereal's charge mechanic is unique to their stomache, instead of this really weird inbetween where just nothing happens and you kinda starve to death. You can stick an ethereal stomach in a human and this will replace their hunger mechanic with the charge mechanic and lets them absorb power from APCs and everything. You can also give ethereals a normal stomach and they'll use the normal hunger mechanics 

:cl:
refactor: hunger mechanics are now inherent to stomaches
expansion: ethereal charge mechanic is now unique to their stomache and can be transferred freely
/:cl:

### Why
I don't like species code, and I wanna slowly move all species behaviour to be modular to their organs and limbs. It's just that holy shit there's a real long fucking road ahead but this is a start

### CODE REVIEW

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -703,14 +703,13 @@
 		var/mob/living/carbon/human/H = user
 
 		if(istype(H))
-			var/datum/species/ethereal/eth_species = H.dna?.species
-			if(istype(eth_species))
-				var/datum/species/ethereal/E = H.dna.species
-				var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
-				if((E.drain_time > world.time) || !stomach)
+			var/obj/item/organ/stomach/maybe_stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
+			if(istype(maybe_stomach, /obj/item/organ/stomach/ethereal))
```
Comment (Time-Green) [inline code/modules/power/lighting.dm:707]: There are still a bunch of shitty istype checks like this, but I dont think they're significant enough to be worth reworking

File: code/modules/surgery/organs/stomach/stomach_ethereal.dm [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	name = "biological battery"
+	icon_state = "stomach-p" //Welp. At least it's more unique in functionaliy.
+	desc = "A crystal-like organ that stores the electric charge of ethereals."
+	///basically satiety but electrical
+	var/crystal_charge = ETHEREAL_CHARGE_FULL
+	///used to keep ethereals from spam draining power sources
+	var/drain_time = 0
+
+/obj/item/organ/stomach/ethereal/on_life(delta_time, times_fired)
+	..()
```
Comment (Rohesie) [inline code/modules/surgery/organs/stomach/stomach_ethereal.dm:11]:
```suggestion
	. = ..()
```
Preserve the parent return unless you have a specific reason not to.

File: code/modules/surgery/organs/stomach/stomach_ethereal.dm [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	adjust_charge(-ETHEREAL_CHARGE_FACTOR * delta_time)
+	handle_charge(owner, delta_time, times_fired)
+
+/obj/item/organ/stomach/ethereal/Insert(mob/living/carbon/carbon, special = 0)
+	..()
```
Comment (Rohesie) [inline code/modules/surgery/organs/stomach/stomach_ethereal.dm:16]:
```suggestion
	. = ..()
```

File: code/modules/surgery/organs/stomach/stomach_ethereal.dm [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+/obj/item/organ/stomach/ethereal/Remove(mob/living/carbon/carbon, special = 0)
+	UnregisterSignal(owner, COMSIG_PROCESS_BORGCHARGER_OCCUPANT)
+	UnregisterSignal(owner, COMSIG_LIVING_ELECTROCUTE_ACT)
+	REMOVE_TRAIT(owner, TRAIT_NOHUNGER, src)
+
+	carbon.clear_alert("ethereal_charge")
+	carbon.clear_alert("ethereal_overcharge")
+
+	..()
```
Comment (Rohesie) [inline code/modules/surgery/organs/stomach/stomach_ethereal.dm:29]:
```suggestion
	return ..()
```

**Comment by Time-Green** [comment]
Thank you for your swift review! There was no reason I didn't return parent procs, vscode just autocompleted that shit and I went with it

**Pull request merged**